### PR TITLE
fix(core/spell): Stop `Blink` from the falling player to move upward by 1y and stop moving backward when facing a wall

### PR DESCRIPTION
**Pull request by Synful-Syn**
## Changes Proposed:
- Using Blink while falling at a high height will no longer warp the player by 1y upward.
- Added a logic that prevents the player from moving backward by using `Blink`.

## Issues Addressed:

- Closes nothing, I've discovered it while failing to correct something else in Blink.

## Tests Performed:

- 
- 

## How to Test the Changes:

Learn Blink: `.learn 1953`

Use Blink without having to wait for the mana or the cooldowns.
`.cheat cooldown`
`.mod mana 239432432`

Make a `.gps` macro so you may see your Z coordinate while testing the falling from a high altitude.

## Known Issues and TODO List:

-

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -1586,6 +1586,15 @@ void Spell::SelectImplicitCasterDestTargets(SpellEffIndex effIndex, SpellImplici
                                 destz = prevZ;
                             //LOG_ERROR("spells", "(collision) destZ rewrited in prevZ");
 
+                            // Don't make the player move backward from the xy adjustments by collisions.
+                            if ((DELTA_X > 0 && startx > destx) || (DELTA_X < 0 && startx < destx)
+                            ||  (DELTA_Y > 0 && starty > desty) || (DELTA_Y < 0 && starty < desty))
```
Comment (Kitzunu) [inline src/server/game/Spells/Spell.cpp:1591]:
```suggestion
                            if ((DELTA_X > 0 && startx > destx) || (DELTA_X < 0 && startx < destx)
                                ||  (DELTA_Y > 0 && starty > desty) || (DELTA_Y < 0 && starty < desty))
```

Comment (Synful-Syn) [reply src/server/game/Spells/Spell.cpp:1591]: I made the spacing this way so they would be perfectly alligned for readability purposes.
image

With the change you are offering, there would be a double space right here.
image

I do prefer not changing it but if we must, I'd remove the double space or making it a single hard-to-read line.
image

What do you prefer? Making it a single line or removing the double space?

Comment (Kitzunu) [reply src/server/game/Spells/Spell.cpp:1591]: Remove the double space, I had missed that initially

Comment (Kitzunu) [reply src/server/game/Spells/Spell.cpp:1591]: What you could do is moeve the || to the previous line for readability

Comment (Synful-Syn) [reply src/server/game/Spells/Spell.cpp]: I did not think of that, I love it. :D

**Review by Nefertumm — APPROVED** [review]

**Comment by GastonCK** [comment]
Tested
https://www.youtube.com/watch?v=tukMsHaJV50

**Pull request merged**